lib/mix/lib/mix/utils.ex

```diff
@@ -549,4 +549,40 @@ defmodule Mix.Utils do
   defp url?(path) do
     URI.parse(path).scheme in ["http", "https"]
   end
+
+  @doc """
+  Checks if two strings are similar based on their Levenshtein distance
+  """
+  def similar?(source, target, max_distance \\ 2) do
+    distance(to_char_list(source), to_char_list(target)) < max_distance
+  end
+
+  # The algorithm is based on the Wagner-Fischer iterative implementation
+  # of Levenshtein algorithm.
+  defp distance(source, source), do: 0
+
+  defp distance(source, []), do: length(source)
+
+  defp distance([], target), do: length(target)
+
+  defp distance(source, target) do
+    distlist = 0..length(target) |> Enum.to_list
+    do_distance(source, target, distlist, 1)
+  end
+
+  defp do_distance([], _, distlist, _), do: List.last(distlist)
+
+  defp do_distance([src_hd | src_tl], target, distlist, step) do
+    distlist = distlist(target, distlist, src_hd, [step], step)
+    do_distance(src_tl, target, distlist, step + 1)
+  end
+
+  defp distlist([], _, _, new_distlist, _), do: Enum.reverse(new_distlist)
+
+  defp distlist([target_hd | target_tl], [distlist_hd | distlist_tl],
+                char, new_distlist, last_dist) when length(distlist_tl) > 0 do
+    diff = if target_hd != char, do: 1, else: 0
+    min = Enum.min([last_dist + 1, hd(distlist_tl) + 1, distlist_hd + diff])
```

Comment:
Gah, sorry. It could be `min(min(...), ...)` then as it would skip the overhead of going through the list. It doesn't matter if the code remains here but it would if we want to move it to String.